app/test/features/chat/custom_input.dart

```diff
@@ -212,10 +260,60 @@ void main() {
         await tester.pump();
         await tester.enterTextWithoutReplace(find.byType(TextField), 't');
 
-        await tester.pump();
+        // This test is timing out due to a pending timer.
+        // See MultiTriggerAutocompleteState._onChangedField in:
+        // acter_trigger_autocomplete.dart:279
+        // put 300ms delay as (debounceTimerDuration)
+        await tester.pump(Durations.medium2);
         expect(controller.text, 'testing code'); // <- cursor moved
       },
-      skip: true,
+    );
+
+    testWidgets(
+      'Edit message shows correct message state in controller',
+      (tester) async {
+        await tester.pumpWidget(
+          ProviderScope(
+            overrides: [
+              sdkProvider.overrideWith((ref) => MockActerSdk()),
+              ...overrides,
+            ],
+            child: const InActerContextTestWrapper(
+              child: CustomChatInput(
+                roomId: 'roomId',
+              ),
+            ),
+          ),
+        );
+        // not visible
+        expect(find.byKey(CustomChatInput.noAccessKey), findsNothing);
+        expect(find.byKey(CustomChatInput.loadingKey), findsNothing);
+
+        final element = tester.element(find.byType(CustomChatInput));
+        final container = ProviderScope.containerOf(element);
+
+        final TextField textField = tester.widget(find.byType(TextField));
+        final controller = textField.controller!;
+
+        // initial state should be empty
+        assert(controller.text.trim().isEmpty, true);
+
```

Comment:
Can I also has a test that confirms if it wasn't empty and we route to another chat it is not the same and come back, it holds the previous state for that same room? That's the main concern I have with this changeset and a test proofing it works would relieve my worry.

Reply:
The new refactored mentions code (with text controller) isn't storing any previous state of room when switching. As I wanted to switch this mechanism with ```compose draft``` in the follower PRs. Do you think we want to switch it back to riverpod version of state storing for now? And migrate that to ```composeDraft``` in separate PR ? OR wait ? I will test in both cases...

Reply:
well, could you add a test and have it skipped for now? and we'll fix it in the composedraft one? but then this second one needs to come soon .... before the next release ideally ...